src/main/host/descriptor/tcp.c

```diff
@@ -1551,9 +1551,9 @@ void tcp_getInfo(TCP* tcp, struct tcp_info *tcpinfo) {
 //  tcpinfo->tcpi_rto;
 //  tcpinfo->tcpi_ato;
     tcpinfo->tcpi_snd_mss =
-        (u_int32_t)(CONFIG_MTU - CONFIG_HEADER_SIZE_TCPIP - CONFIG_HEADER_SIZE_ETH);
+        (u_int32_t)(CONFIG_MTU - CONFIG_HEADER_SIZE_TCPIP);
```

Comment:
Consider using the CONFIG_TCP_MAX_SEGMENT_SIZE constant defined in src/main/core/support/definitions.h.